content/300-guides/050-database/900-turso.mdx

```diff
@@ -0,0 +1,202 @@
+
+## Connect and query your primary database
+
+<br />
+
+<details>
+
+<summary>
+  Expand to learn how to provision a database and retrieve your credentials on
+  Turso
+</summary>
+
+Ensure that you have the [Turso CLI](https://docs.turso.tech/reference/turso-cli) installed to manage your databases.
+
+1. To provision a database on Turso, run the following command:
+
+```terminal
+turso db create turso-prisma-db
+```
+
+2. Retrieve the database's connection string:
+
+```terminal
+turso db show turso-prisma-db
+```
+
+3. Create an authentication token that will allow you to connect to the database:
+
+```terminal
+turso db tokens create turso-prisma-db
+```
+
+4. Update your `.env` file with the authentication token and connection string:
+
+```text file=.env
+TURSO_AUTH_TOKEN="eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9..."
+TURSO_DATABASE_URL="libsql://turso-prisma-db-user.turso.io"
+```
+
+</details>
+
+To get started, enable the `driverAdapters` Preview feature flag:
+
+```prisma highlight=3;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+
+datasource db {
+  provider = "sqlite"
+  url      = "file:./dev.db"
+}
+```
+
+Generate Prisma Client:
+
+```terminal
+npx prisma generate
+```
+
+Install the libSQL database client and Prisma driver adapter for libSQL packages:
+
+```terminal
+npm install @libsql/client @prisma/adapter-libsql
+```
+
+Update your Prisma Client instance:
+
+```ts
+import { PrismaClient } from '@prisma/client'
+import { PrismaLibSQL } from '@prisma/adapter-libsql'
+import { createClient } from '@libsql/client'
+
+const libsql = createClient({
+  url: `${process.env.TURSO_DATABASE_URL}`,
+  authToken: `${process.env.TURSO_AUTH_TOKEN}`,
+})
+
+const adapter = new PrismaLibSQL(libsql)
+const prisma = new PrismaClient({ adapter })
+```
+
+You can use Prisma Client as you normally would with full type-safety in your project.
+
+## How to manage schema changes
+
+Prisma Migrate and Introspection workflows are currently not supported when working with Turso. This is because Turso uses HTTP to connect to your database, which Prisma Migrate doesn't support.
+
+To update your database schema:
+
+1. Generate a migration file using `prisma migrate dev` against a local SQLite database:
+
+   ```terminal
+   npx prisma migrate dev --name init # Migration name
+   ```
+
+2. Apply the migration using Turso's CLI:
+
+   ```terminal
+   turso db shell turso-prisma-db < ./prisma/migrations/20230922132717_init/migration.sql # Replace `20230922132717_init` with the existing migration
+   ```
+
+For subsequent migrations, repeat the above steps to apply changes to your database. This workflow does not support tracking the history of applied migrations.
```

Comment:
> This workflow does not support tracking the history of applied migrations.

That is exactly what `migrate dev` does.
Do you mean that it does not track what was applied to the remote database?

Reply:
> Do you mean that it does not track what was applied to the remote database?

Yes.

Reply:
I added the "remote database" to the sentence.